framework/finalizer_test.go

```diff
@@ -0,0 +1,117 @@
+package framework
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/micrologger"
+
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest/fake"
+	"k8s.io/kubernetes/pkg/api/testapi"
+)
+
+func Test_addFinalizer(t *testing.T) {
+	testCases := []struct {
+		name           string
+		object         *apiv1.Pod
+		operatorName   string
+		expectedResult bool
+		errorMatcher   func(error) bool
+	}{
+		{
+			name: "case 0: No finalizer is set yet",
+			object: &apiv1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "TestPod",
+					Namespace: "TestNamespace",
+				},
+			},
+			operatorName:   "test-operator",
+			expectedResult: true,
+			errorMatcher:   nil,
+		},
+		{
+			name: "case 1: Finalizer is already set",
+			object: &apiv1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Finalizers: []string{"operatorkit.giantswarm.io/test-operator"},
+					Name:       "TestPod",
+					Namespace:  "TestNamespace",
+				},
+			},
+			operatorName:   "test-operator",
+			expectedResult: false,
+			errorMatcher:   nil,
+		},
+	}
+	logger, err := micrologger.New(micrologger.DefaultConfig())
+	if err != nil {
+		t.Fatalf("micrologger.New() failed: %#v", err)
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			c := &fake.RESTClient{
```

Comment:
This is the only way I got this to work. There is no `fake.NewSimpleRESTClient` like there is for the k8sclient, a fake HTTP client has to be passed, otherwise this would not work for me.